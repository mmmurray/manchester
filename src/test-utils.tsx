import * as RTL from '@testing-library/react'
import React, { FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from './theme'

const Wrapper: FC = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children as any}</ThemeProvider>
)

const render = (el: React.ReactElement) => RTL.render(<Wrapper>{el}</Wrapper>)

export * from '@testing-library/react'
export { render, Wrapper }
