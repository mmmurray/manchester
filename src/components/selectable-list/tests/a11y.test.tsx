import React from 'react'
import SelectableList from '..'
import { Wrapper } from '../../../test-utils'

test('default selectable list a11y', async () => {
  await expect(
    <Wrapper>
      <SelectableList
        items={[
          { id: '1', title: 'Item 1' },
          { id: '2', title: 'Item 2' },
          { id: '3', title: 'Item 3' },
        ]}
      />
    </Wrapper>,
  ).toMatchA11ySnapshot()
})

test('selectable list with button accessories a11y', async () => {
  await expect(
    <Wrapper>
      <SelectableList
        items={[
          {
            id: '1',
            title: 'Item 1',
            accessories: [
              <button key={1} type="button">
                B1
              </button>,
            ],
          },
          {
            id: '2',
            title: 'Item 2',
            accessories: [
              <button key={1} type="button">
                B2
              </button>,
            ],
          },
          {
            id: '3',
            title: 'Item 3',
            accessories: [
              <button key={1} type="button">
                B3
              </button>,
            ],
          },
        ]}
      />
    </Wrapper>,
  ).toMatchA11ySnapshot()
})
