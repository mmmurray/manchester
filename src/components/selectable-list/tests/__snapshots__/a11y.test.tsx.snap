// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default selectable list a11y: toMatchA11ySnapshot 1`] = `
"role: WebArea
children:
  -
    role: list
    children:
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 1'
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 2'
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 3'
"
`;

exports[`selectable list with button accessories a11y: toMatchA11ySnapshot 1`] = `
"role: WebArea
children:
  -
    role: list
    children:
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 1'
          -
            role: button
            name: B1
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 2'
          -
            role: button
            name: B2
      -
        role: listitem
        children:
          -
            role: button
            name: 'Item 3'
          -
            role: button
            name: B3
"
`;
